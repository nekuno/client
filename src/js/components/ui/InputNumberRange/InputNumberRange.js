import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { createSliderWithTooltip } from 'rc-slider';
import Slider from 'rc-slider';
const Range = createSliderWithTooltip(Slider.Range);
import styles from './InputNumberRange.scss';
import 'rc-slider/assets/index.css';

export default class InputNumberRange extends Component {

    static propTypes = {
        title  : PropTypes.string,
        value  : PropTypes.array,
        minNum: PropTypes.number.isRequired,
        maxNum: PropTypes.number.isRequired,
        onChangeHandler : PropTypes.func.isRequired
    };

    handleChangeRange(range) {
        this.props.onChangeHandler(range);
    }

    render() {
        const {value, minNum, maxNum} = this.props;
        let rangeProps = {
            marks: {[minNum]: minNum, [Math.round((maxNum + minNum) / 2)]: Math.round((maxNum + minNum) / 2), [maxNum]: maxNum},
            min: minNum,
            max: maxNum,
            defaultValue: value,
            handleStyle: [{background: '#756EE5', color: '#756EE5', borderColor: '#756EE5', boxShadow: 'none'}],
            trackStyle: [{background: '#756EE5', color: '#756EE5', borderColor: '#756EE5'}],
            dotStyle: {background: 'transparent', borderColor: 'transparent'},
            activeDotStyle: {background: 'transparent', borderColor: 'transparent'},
            onAfterChange: this.handleChangeRange.bind(this)
        };
        if (!value) {
            rangeProps.value = [minNum, minNum];
        }

        return (
            <div className={styles.inputNumberRange}>
                <div className={styles.title + ' small'}>
                    {this.props.title}
                </div>
                <div className={styles.range}>
                    <Range {...rangeProps}/>
                </div>
            </div>
        );
    }
}