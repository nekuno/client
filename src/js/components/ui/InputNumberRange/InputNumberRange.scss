@import '../../../../scss/_partials/colors';

.input-number-range {

  .title {
    font-weight: 600;
  }
  .range {
    margin-top: 16px;
    margin-bottom: 48px;
    padding-left: 16px;
    padding-right: 48px;
  }
}