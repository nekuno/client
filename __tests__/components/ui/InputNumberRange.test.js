import React from 'react';
import InputNumberRange from '../../../src/js/components/ui/InputNumberRange/InputNumberRange.js';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test InputRange component', () => {
    const mockFn = jest.fn();
    const inputRange = shallow(<InputNumberRange minNum={8} maxNum={80} onChangeHandler={mockFn} value={[8, 11]}/>);

    it('should be defined', () => {
        expect(inputRange).toBeDefined();
    });
    it('should render correctly', () => {
        expect(inputRange).toMatchSnapshot();
    });
});
