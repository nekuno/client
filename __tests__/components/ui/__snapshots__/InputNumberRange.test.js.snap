// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InputRange component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputNumberRange
    maxNum={80}
    minNum={8}
    onChangeHandler={[MockFunction]}
    value={
      Array [
        8,
        11,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title small"
        />,
        <div
          className="range"
        >
          <ComponentWrapper
            activeDotStyle={
              Object {
                "background": "transparent",
                "borderColor": "transparent",
              }
            }
            defaultValue={
              Array [
                8,
                11,
              ]
            }
            dotStyle={
              Object {
                "background": "transparent",
                "borderColor": "transparent",
              }
            }
            handleStyle={
              Array [
                Object {
                  "background": "#756EE5",
                  "borderColor": "#756EE5",
                  "boxShadow": "none",
                  "color": "#756EE5",
                },
              ]
            }
            marks={
              Object {
                "44": 44,
                "8": 8,
                "80": 80,
              }
            }
            max={80}
            min={8}
            onAfterChange={[Function]}
            tipFormatter={[Function]}
            tipProps={Object {}}
            trackStyle={
              Array [
                Object {
                  "background": "#756EE5",
                  "borderColor": "#756EE5",
                  "color": "#756EE5",
                },
              ]
            }
          />
        </div>,
      ],
      "className": "inputNumberRange",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "title small",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ComponentWrapper
            activeDotStyle={
              Object {
                "background": "transparent",
                "borderColor": "transparent",
              }
            }
            defaultValue={
              Array [
                8,
                11,
              ]
            }
            dotStyle={
              Object {
                "background": "transparent",
                "borderColor": "transparent",
              }
            }
            handleStyle={
              Array [
                Object {
                  "background": "#756EE5",
                  "borderColor": "#756EE5",
                  "boxShadow": "none",
                  "color": "#756EE5",
                },
              ]
            }
            marks={
              Object {
                "44": 44,
                "8": 8,
                "80": 80,
              }
            }
            max={80}
            min={8}
            onAfterChange={[Function]}
            tipFormatter={[Function]}
            tipProps={Object {}}
            trackStyle={
              Array [
                Object {
                  "background": "#756EE5",
                  "borderColor": "#756EE5",
                  "color": "#756EE5",
                },
              ]
            }
          />,
          "className": "range",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeDotStyle": Object {
              "background": "transparent",
              "borderColor": "transparent",
            },
            "defaultValue": Array [
              8,
              11,
            ],
            "dotStyle": Object {
              "background": "transparent",
              "borderColor": "transparent",
            },
            "handleStyle": Array [
              Object {
                "background": "#756EE5",
                "borderColor": "#756EE5",
                "boxShadow": "none",
                "color": "#756EE5",
              },
            ],
            "marks": Object {
              "44": 44,
              "8": 8,
              "80": 80,
            },
            "max": 80,
            "min": 8,
            "onAfterChange": [Function],
            "tipFormatter": [Function],
            "tipProps": Object {},
            "trackStyle": Array [
              Object {
                "background": "#756EE5",
                "borderColor": "#756EE5",
                "color": "#756EE5",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title small"
          />,
          <div
            className="range"
          >
            <ComponentWrapper
              activeDotStyle={
                Object {
                  "background": "transparent",
                  "borderColor": "transparent",
                }
              }
              defaultValue={
                Array [
                  8,
                  11,
                ]
              }
              dotStyle={
                Object {
                  "background": "transparent",
                  "borderColor": "transparent",
                }
              }
              handleStyle={
                Array [
                  Object {
                    "background": "#756EE5",
                    "borderColor": "#756EE5",
                    "boxShadow": "none",
                    "color": "#756EE5",
                  },
                ]
              }
              marks={
                Object {
                  "44": 44,
                  "8": 8,
                  "80": 80,
                }
              }
              max={80}
              min={8}
              onAfterChange={[Function]}
              tipFormatter={[Function]}
              tipProps={Object {}}
              trackStyle={
                Array [
                  Object {
                    "background": "#756EE5",
                    "borderColor": "#756EE5",
                    "color": "#756EE5",
                  },
                ]
              }
            />
          </div>,
        ],
        "className": "inputNumberRange",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "title small",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ComponentWrapper
              activeDotStyle={
                Object {
                  "background": "transparent",
                  "borderColor": "transparent",
                }
              }
              defaultValue={
                Array [
                  8,
                  11,
                ]
              }
              dotStyle={
                Object {
                  "background": "transparent",
                  "borderColor": "transparent",
                }
              }
              handleStyle={
                Array [
                  Object {
                    "background": "#756EE5",
                    "borderColor": "#756EE5",
                    "boxShadow": "none",
                    "color": "#756EE5",
                  },
                ]
              }
              marks={
                Object {
                  "44": 44,
                  "8": 8,
                  "80": 80,
                }
              }
              max={80}
              min={8}
              onAfterChange={[Function]}
              tipFormatter={[Function]}
              tipProps={Object {}}
              trackStyle={
                Array [
                  Object {
                    "background": "#756EE5",
                    "borderColor": "#756EE5",
                    "color": "#756EE5",
                  },
                ]
              }
            />,
            "className": "range",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeDotStyle": Object {
                "background": "transparent",
                "borderColor": "transparent",
              },
              "defaultValue": Array [
                8,
                11,
              ],
              "dotStyle": Object {
                "background": "transparent",
                "borderColor": "transparent",
              },
              "handleStyle": Array [
                Object {
                  "background": "#756EE5",
                  "borderColor": "#756EE5",
                  "boxShadow": "none",
                  "color": "#756EE5",
                },
              ],
              "marks": Object {
                "44": 44,
                "8": 8,
                "80": 80,
              },
              "max": 80,
              "min": 8,
              "onAfterChange": [Function],
              "tipFormatter": [Function],
              "tipProps": Object {},
              "trackStyle": Array [
                Object {
                  "background": "#756EE5",
                  "borderColor": "#756EE5",
                  "color": "#756EE5",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
